src/ivoc/symdir.cpp

```diff
@@ -218,11 +215,10 @@ SymDirectory::~SymDirectory() {
     delete impl_;
 }
 void SymDirectoryImpl::disconnect(Observable*) {
-    long cnt = symbol_list_.count();
-    for (long i = 0; i < cnt; ++i) {
-        delete symbol_list_.item(i);
+    for (auto& item: symbol_lists_) {
+        delete item;
     }
-    symbol_list_.remove_all();
+    symbol_lists_.clear();
```

Comment:
Do we want to release the memory in the `std::vector`? If yes, we need `shrink_to_fit` or swap.